src/iccpd/include/mlacp_tlv.h

```diff
@@ -0,0 +1,455 @@
+    uint16_t partner_sys_priority;
+    /* [IEEE-802.1AX], Section 5.4.2.2, item s. */
+    uint16_t partner_key;
+    /* [IEEE-802.1AX], Section 5.4.6. */
+    uint16_t agg_id;
+    /* [IEEE-802.1AX], Section 5.3.5. */
+    uint16_t actor_key;
+    /*
+     * RFC 7275
+     * 7.2.8.  mLACP Aggregator State TLV
+     * [Page 61]
+     */
+    uint8_t agg_state;
+} __attribute__ ((packed));
+
+typedef struct mLACPAggPortStateTLV mLACPAggPortStateTLV;
+
+/*
+ * RFC 7275
+ * 7.2.9.  mLACP Synchronization Request TLV
+ * [Page 61]
+ */
+struct mLACPSyncReqTLV
+{
+    ICCParameter icc_parameter;
+    /*
+     * RFC 7275
+     * 7.2.9.  mLACP Synchronization Request TLV
+     * [Page 62]
+     */
+    uint16_t req_num;
+
+#if __BYTE_ORDER == __BIG_ENDIAN
+    uint16_t c_bit : 1;
+    /*
+     * RFC 7275
+     * 7.2.9.  mLACP Synchronization Request TLV
+     * [Page 63]
+     */
+    uint16_t s_bit : 1;
+    uint16_t req_type : 14;
+#elif __BYTE_ORDER == __LITTLE_ENDIAN
+    uint16_t req_type : 14;
+    /*
+     * RFC 7275
+     * 7.2.9.  mLACP Synchronization Request TLV
+     * [Page 63]
+     */
+    uint16_t s_bit : 1;
+    uint16_t c_bit : 1;
+#endif
+    /* [IEEE-802.1AX], Section 5.3.4. */
+    /* [IEEE-802.1AX], Section 5.4.6. */
+    uint16_t port_num_agg_id;
+    /* [IEEE-802.1AX], Section 5.3.5. */
+    uint16_t actor_key;
+} __attribute__ ((packed));
+
+typedef struct mLACPSyncReqTLV mLACPSyncReqTLV;
+
+/*
+ * RFC 7275
+ * 7.2.10.  mLACP Synchronization Data TLV
+ * [Page 63]
+ */
+struct mLACPSyncDataTLV
+{
+    ICCParameter icc_parameter;
+    /*
+     * RFC 7275
+     * 7.2.10.  mLACP Synchronization Data TLV
+     * [Page 64]
+     */
+    uint16_t req_num;
+    uint16_t flags;
+} __attribute__ ((packed));
+
+typedef struct mLACPSyncDataTLV mLACPSyncDataTLV;
+
+/* VLAN Information TLV*/
+struct mLACPVLANData
+{
+    uint16_t vlan_id;
+} __attribute__ ((packed));
+
+/*
+ * Port Channel Information TLV
+ */
+struct mLACPPortChannelInfoTLV
+{
+    ICCParameter icc_parameter;
+    uint16_t agg_id;
+    char if_name[MAX_L_PORT_NAME];
+    uint8_t if_name_len;
+    uint8_t l3_mode;
+    uint32_t ipv4_addr;
+    uint16_t orphan;
+    uint16_t po_id;
+    uint16_t num_of_vlan_id;
+    struct mLACPVLANData vlanData[0];
+} __attribute__ ((packed));
+
+typedef struct mLACPPortChannelInfoTLV mLACPPortChannelInfoTLV;
+
+/*
+ * Port PeerLink Information TLV
+ */
+struct mLACPPeerLinkInfoTLV
+{
+    ICCParameter icc_parameter;
+    char if_name[MAX_L_PORT_NAME];
+    uint8_t port_type;
+} __attribute__ ((packed));
+
+typedef struct mLACPPeerLinkInfoTLV mLACPPeerLinkInfoTLV;
+
+struct mLACPVLANInfoTLV
+{
+    ICCParameter icc_parameter;
+    uint16_t id; /* Local Interface ID, not VLAN ID */
+    uint16_t num_of_vlan_id;
+    struct mLACPVLANData vlanData[0];
+} __attribute__ ((packed));
+
+/*
+ * MAC Information TLV
+ */
+struct mLACPMACInfoTLV
+{
+    ICCParameter    icc_parameter;
+    uint8_t         type;/*add or del*/
+    char     mac_str[32];
```

Comment:
For each MAC, we should send only 6-byte MAC address instead of the 32-byte MAC string to MLAG peer.
For L2 MCLAG, large number of MACs will be synched between MLAG peers. Hence, we plan to make the change as part of the L2 MCLAG enhancements and will contribute back to the community.

Reply:
Your proposal is reasonable. This can be refine to 18-byte string. If define to 6-bytes, the receiver must transfer it to 18-byte MAC string. This is CPU consuming.